### PR TITLE
Replace nginx with istio

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Migration of Application Connector from Nginx to Istio. 

Changes proposed in this pull request:

- Added Ingress Gateway CA secret
- Replaced Ingresses with Virtual Services
- Modified Gateway
- Updated Application Connector chart
- Updated images

Documentation changes: #4246

**Related issue(s)**
#1608

### CODE REVIEW

File: resources/application-connector/charts/connectivity-certs-controller/README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ The Connectivity Certs Controller fetches the client certificate and the root CA
 You can set the following parameters in the Connectivity Certs Controller chart:
 - **appName** - Name of the Controller used during registration. The default value is `connectivity-certs-controller`.
 - **namespace** - Namespace in which the Controller creates the Secrets that store certificates. The default Namespace is `kyma-integration`.
-- **clusterCertificatesSecret** - Name of the Secret which stores the client certificate and key. The default name is `cluster-client-certificates`.
-- **caCertificatesSecret** - Name of the Secret which stores the CA certificate. The default name is `nginx-auth-ca`.
+- **clusterCertificatesSecret** - Namespace and Name of the Secret which stores the client certificate and key in the format `Namespace/Name`. The default name is `kyma-integration/cluster-client-certificates`.
```
Comment (tomekpapiernik) [inline resources/application-connector/charts/connectivity-certs-controller/README.md]:
```suggestion
- **clusterCertificatesSecret** - Namespace and the name of the Secret which stores the client certificate and key. Requires in the `Namespace/secret name` format. The default value is `kyma-integration/cluster-client-certificates`.
```

File: resources/application-connector/charts/connectivity-certs-controller/README.md [hunk 2]
```diff
@@ -7,8 +7,8 @@ The Connectivity Certs Controller fetches the client certificate and the root CA
...
+- **caCertificatesSecret** - Namespace and Name of the Secret which stores the CA certificate in the format `Namespace/Name`. The default name is `istio-system/application-connector-ca-certs`.
```
Comment (tomekpapiernik) [inline resources/application-connector/charts/connectivity-certs-controller/README.md]:
```suggestion
- **caCertificatesSecret** - Namespace and the name of the Secret which stores the client certificate and key. Requires the `Namespace/secret name` format. The default value is `istio-system/application-connector-ca-certs`.
```

File: resources/application-connector/charts/connector-service/README.md [hunk 3]
```diff
@@ -14,7 +14,8 @@ The Connector Service has the following parameters, that can be set through the
 - **tokenLength** - Length of registration tokens. The default value is `64`.
 - **appTokenExpirationMinutes** - Time after which tokens for Applications expire and are no longer valid. The default value is `5` minutes.
 - **runtimeTokenExpirationMinutes** - Time after which tokens for Runtimes expire and are no longer valid. The default value is `10` minutes.
-- **caSecretName** - Name of the Secret which contains the root Certificate Authority (CA). The default value is `connector-service-app-ca`.
+- **caSecretName** - Namespace and Name of the secret, in Namespace/Name format, which contains the certificate and key used for signing client certificates. The default value is `kyma-integration/application-connector-ca-certs`.
```
Comment (tomekpapiernik) [inline resources/application-connector/charts/connector-service/README.md]:
```suggestion
- **caSecretName** - Namespace and the name of the Secret which stores the certificate and key used to sign client certificates. Requires the `Namespace/secret name` format. The default value is `kyma-integration/application-connector-ca-certs`.
```

File: resources/application-connector/charts/connector-service/README.md [hunk 4]
```diff
@@ -14,7 +14,8 @@ The Connector Service has the following parameters, that can be set through the
...
+- **rootCACertificateSecretName** - Namespace and Name of the secret, in Namespace/Name format, which contains root CA (Certificate Authority) Certificate in case certificates are singed by intermediate CA.
```
Comment (tomekpapiernik) [inline resources/application-connector/charts/connector-service/README.md]:
```suggestion
- **rootCACertificateSecretName** - Namespace and the name of the Secret which stores the root CA (Certificate Authority) Certificate. Requires the `Namespace/secret name` format.
```

Comment (tomekpapiernik) [reply resources/application-connector/charts/connector-service/README.md]: There's no default value for this one, is this by design?

Comment (Szymongib) [reply resources/application-connector/charts/connector-service/README.md]: The default is empty, as this parameter is not required.

File: resources/application-connector/charts/connectivity-certs-controller/README.md (hunk 2 above)
Comment (Szymongib) [reply resources/application-connector/charts/connectivity-certs-controller/README.md]: Changed it to `- **caCertificatesSecret** - Namespace and the name of the Secret which stores the CA certificate. Requires the `Namespace/secret name` format. The default value is `istio-system/application-connector-ca-certs`.` as it does store CA certificate, not the client one.

**Pull request merged**